tests/pytorch/test_stream.py

```diff
@@ -34,6 +62,64 @@ def test_basics():
     gg = g.to(device=F.ctx())
     OPS.copy_u_sum(gg, xx)
 
+@unittest.skipIf(F._default_context_str == 'cpu', reason="stream only runs on GPU.")
+def test_set_get_stream():
+    current_stream = torch.cuda.current_stream()
+    # test setting another stream
+    s = torch.cuda.Stream(device=F.ctx())
+    torch.cuda.set_stream(s)
+    assert to_dgl_stream_handle(s).value == _dgl_get_stream(to_dgl_context(F.ctx())).value
+    # revert to default stream
+    torch.cuda.set_stream(current_stream)
+
+@unittest.skipIf(F._default_context_str == 'cpu', reason="stream only runs on GPU.")
+def test_record_stream_smoke():
+    # smoke test to ensure it can work
+    stream = torch.cuda.Stream()
+    with torch.cuda.stream(stream):
+        t = nd.array(np.array([1., 2., 3., 4.]), ctx=nd.gpu(0))
+        g = rand_graph(10, 20, device=F.cuda())
+    # NDArray is an internal object, just use DGLStreamHandle
+    t.record_stream(to_dgl_stream_handle(torch.cuda.current_stream()))
+    g.record_stream(torch.cuda.current_stream())
+
+@unittest.skipIf(F._default_context_str == 'cpu', reason="stream only runs on GPU.")
+# borrowed from PyTorch, test/test_cuda.py: test_record_stream()
+def test_record_stream():
```

Comment:
do we need to test `record_stream` for dgl graph object other than smoke testing?

Reply:
A blocker is that we don't have an equivalent API for dgl.graph as `tensor.data_ptr()` to check if the memory has been reused or not. The underlying sparse matrix or NDArrays of dgl.graph have not been exposed to Python and we have not way to access them.

Reply:
I see... Can you run a different test other than directly checking the allocator? For example, checking spmm results like
```
# default stream
g = rand_dgl_graph.to(device) # g allocates on gpu with default stream
copy_stream = torch.cuda.Stream()
with copy_stream:
       g2 = g.to(device)  # copy g on copy_stream
       sleep(100000)   # make copy takes very long time
g.record_stream(copy_stream)  # necessary to ensure copy is finished
g = dgl_unit_graph # access and modify g on default stream again 
res = copy_u_sum(g, x)  # check if result is correct
```

Reply:
Good idea. I have added both positive and negative tests.